### DISCUSSION TITLE
유니티 신(포탈) 이동시 멀티플레이 질문

### DISCUSSION THREAD

**Discussion by bonokbs** [naverz/zepeto-studio-kor#685] [5. Q&A]
안녕하세요.
맵(Jump map) 하고 일반맵(daylight_2map) 으로 구성되어있습니다.

![image](https://user-images.githubusercontent.com/32977809/166118158-7c839b80-08f2-4e83-bdb6-3c6e5eb20081.png)
![image](https://user-images.githubusercontent.com/32977809/166118200-c8d10fc8-8aae-4b76-be6f-a13213aa2438.png)

2개를 자유롭게 이동하려고 포탈기능을 구현했습니다.
SceneManager.LoadScene("night_4map");  명령어로 신을 이동명령을 하니 잘 넘어가는데 문제는

A 사용자가 일반맵에서 게임맵으로 넘어가도 -> B사용자는 일반맵에서 A사용자가 움직임이 보이는데
제페토 멀티플레이 스크립트에서 특정 조건일때만 움직이는 좌표값을 넘겨야하는데 감도 안와서 질문 드립니다.
감사합니다.

**Comment by minjujuu** [comment]
A 사용자가 일반맵에서 게임맵으로 넘어갔을 때, B 사용자가 있는 일반맵 씬에서 A 사용자가 보이지 않아야 하는 것인가요?
그렇다면 지금 올라와있는 멀티플레이 샘플 기준으로 플레이어가 멀티플레이 룸을 떠날 때 
해당 플레이어 캐릭터의 인스턴스를 삭제하도록 되어 있어서, 이 부분 관련 변경이 필요할 것 같습니다!

1. A 사용자가 씬을 이동할 때 서버로 메시지를 전송합니다. 
2. 서버에서 1번의 메시지를 받았을 때 다른 사용자들에게 broadcast로 메시지를 전송합니다. 
3. 2번의 메시지를 받은 다른 사용자들은 A 사용자와 같은 씬에 있지 않다면 A 사용자의 캐릭터 인스턴스를 제거합니다.
